background/services/chain/serial-fallback-provider.ts

```diff
@@ -754,17 +762,39 @@ export default class SerialFallbackProvider extends QuaisJsonRpcProvider {
     if (this.currentProviderIndex >= this.providerCreators.length) {
       this.currentProviderIndex = 0
     }
-
     logger.debug(
       "Reconnecting provider at index",
       this.currentProviderIndex,
       "..."
     )
-
-    this.currentProvider = this.providerCreators[this.currentProviderIndex].creator()
-    await this.resubscribe(this.currentProvider)
-
-    // TODO After a longer backoff, attempt to reset the current provider to 0.
```

Comment:
is the second part of this TODO valid now? i dont see the current provider set to 0

Reply:
This TODO is no longer valid